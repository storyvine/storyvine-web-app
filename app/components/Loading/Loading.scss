.Loading {
    &__Spinner {
        margin: auto;
        display: block;
        top: 40%;
        position: relative;
    }
}