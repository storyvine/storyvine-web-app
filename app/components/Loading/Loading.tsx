import * as React from 'react';
import { Spin } from 'antd';
import s from './Loading.scss';
import { LoadingComponentProps } from 'react-loadable';

interface Props extends Partial<LoadingComponentProps> {
  height?: string | number | undefined;
}

const Loading: React.FunctionComponent<Props> = ({ height = '100%' }: Props) => (
  <div style={{ height }}>
    <Spin size="large" className={s.Loading__Spinner} />
  </div>
);

export default Loading;
