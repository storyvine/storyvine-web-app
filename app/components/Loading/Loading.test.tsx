import * as React from 'react';
import { shallow } from 'enzyme';
import Loading from './Loading';

test('Loading renders correctly', () => {
  expect(shallow(<Loading />)).toMatchSnapshot();
});
