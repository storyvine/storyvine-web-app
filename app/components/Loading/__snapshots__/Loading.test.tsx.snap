// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Spin
        className={undefined}
        size="large"
        spinning={true}
        wrapperClassName=""
      />,
      "style": Object {
        "height": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": undefined,
        "size": "large",
        "spinning": true,
        "wrapperClassName": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Spin
          className={undefined}
          size="large"
          spinning={true}
          wrapperClassName=""
        />,
        "style": Object {
          "height": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": undefined,
          "size": "large",
          "spinning": true,
          "wrapperClassName": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
